### ISSUE TITLE
custom ordering enum object has no attribute '_type_definition'

### ISSUE THREAD

**Issue by tasiotas** [blb-ventures/strawberry-django-plus#201]
Hi!

I would like to add third option to ordering enum.

```python
@gql.enum
class ProductOrdering(enum.Enum):
    ASC = "ASC"
    DESC = "DESC"
    NONE = "none"


@gql.django.order(Product)
class ProductOrder:
    slug: gql.auto
    date_published: ProductOrdering
```

Once I run query with this ordering, Im getting this error:

```json
{
  "data": null,
  "errors": [
    {
      "message": "'ProductOrdering' object has no attribute '_type_definition'",
      "locations": [
        {
          "line": 2,
          "column": 3
        }
      ],
      "path": [
        "products"
      ]
    }
  ]
}
```

Could you clues what am I doing wrong?
Thank you 🍓

**Comment by bellini666** [comment]
Hey @tasiotas ,

Can you paste the complete traceback from your terminal for me to understand what might have gone wrong there?

**Comment by tasiotas** [comment]
actually im not getting any error in django terminal, just that response with error

**Comment by bellini666** [comment]
Hrm, I'll try to replicate it in another way.

How does your model look like? Specially regarding that `date_published` field

**Comment by tasiotas** [comment]
thank you for looking into it

```python
class Product(models.Model):
   date_published = models.DateTimeField(null=True, blank=True)
```

**Comment by bellini666** [comment]
@tasiotas ok, I see now what is going on.

First of all, for curiosity, why do you want to add a third option to ordering?

Now explaining the issue: When trying to order the field, [this function](https://github.com/strawberry-graphql/strawberry-graphql-django/blob/main/strawberry_django/ordering.py#L23) gets called.

As you can see in it, it handles only `ASC`/`DESC`, but anything else it considers to be another order type so you can do something [like this](https://strawberry-graphql.github.io/strawberry-graphql-django/references/ordering/#ordering) (note that the `color` passes another object in there)

If you really need this, what you can do is try to monkey patch the `generate_order_args` function with your custom implementation. You might want to define your own `Ordering` enum there so that you can use `gql.auto` and have it work for any field. e.g.

```python
from strawberry_django import ordering

@gql.enum
class Ordering(enum.Enum):
    ASC = "ASC"
    DESC = "DESC"
    NONE = "none"

def generate_order_args(order, prefix=""):
    ... # reimplement this using your Ordering enum and its values

# Monkey patch both
ordering.Ordering = Ordering
ordering.generate_order_args = generate_order_args
```

You have to monkey patch it as early as possible, before any type has been created to avoid the original `Ordering` being created.

If you don't want to override the default Ordering, you can reimplement the existing logic and handle your custom `Ordering` object in the if/elif/else

**Comment by tasiotas** [comment]
thank you for explanation and workarounds. I have just learned monkey patching in Python 💯 

Here is my case.
I am transitioning from Graphene. I have a model that I needs to be sorted based on either `name`, `date_published` or `price` (one at the time, not multiple sort criteria).

So I made an order schema like this.
```python
@gql.django.order(Product)
class ProductOrder:
    name: gql.auto
    date_published: gql.auto
    price: gql.auto
```
This works fine, but I dont know (or its nor possible?) how to make a query where I specify filter key as a variable. Still learning GraphQL concepts.
So I was thinking to have all filters in the query at the same time, but only one of them with `ASC/DESC` value, the rest as `none`.

Well, after your explanation of possible solution, I came up with something else. 
In Graphene I used to send string as a sorting value, then have a custom resolver to handle it. But since this lib currently doesn't support [custom resolvers](https://github.com/strawberry-graphql/strawberry-graphql-django/issues/236) in Ordering, I do it via Filter.

```python
@gql.django.filter(Product, lookups=True)
class ProductFilter:
    ...
    order_by: Optional[str]

    def filter_order_by(self, queryset: QuerySet[Product]):
        return queryset.order_by(self.order_by)

```
